### PR TITLE
Revise CONTRIBUTING.md contents

### LINKED ISSUE #14031: Revise CONTRIBUTING.md contents.
### What version of TiKV are you using?
master

### What operating system and CPU are you using?
macOs, intel

### Steps to reproduce
make build

### What did you expect?
build success

### What did happened?
Script complaining that **protoc** cannot be found and error quit 
image

### What do you suggest
- Explicitly mention protoc as one of the prerequisites to build tikv in CONTRIBUTING.md upfront.
- Fix some typo in the md doc.
- Fix dead hyper links.

### PR DESCRIPTION
**Pull request by woofyzhao**
Signed-off-by: woofyzhao <490467089@qq.com>

Issue Number: close #14031

What's Changed:

- Explicitly mention protoc as one of the prerequisites to build tikv in doc.
- Fix some typo in doc.
- Fix dead hyper links.

Release note
NONE

### CODE REVIEW

**Review by zz-jason — APPROVED** [review]
LGTM

**Comment by woofyzhao** [comment]
Only markdown doc is modified but Jekins CI keeps failing, is this normal?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ To build TiKV you'll need to at least have the following installed:
 * `make` - Build tool (run common workflows)
 * `cmake` - Build tool (required for gRPC)
 * `awk` - Pattern scanning/processing language
+* [`protoc`](https://github.com/protocolbuffers/protobuf/releases) - Google protocol buffer compiler
```
Comment (BusyJay) [inline CONTRIBUTING.md:22]: We have drop the dependency long time ago, I believe it's introduced back because of etcd-client. /cc @YuJuncen

Comment (YuJuncen) [reply CONTRIBUTING.md:22]: Yes, more preciously it is `prost` that brings this requirement. After PD exposing some of the operations over the etcd key space in `pdpb`, I guess we would be able to remove the requirement of etcd soon.

**Review by YuJuncen — APPROVED** [review]
LGTM

**Pull request merged**